server/src/internalClusterTest/java/org/elasticsearch/search/ccs/CrossClusterSearchIT.java

```diff
@@ -0,0 +1,189 @@
+                .setAllowPartialSearchResults(false)
+                .setSize(1000)
+                .execute(),
+            IllegalArgumentException.class,
+            RestStatus.BAD_REQUEST,
+            "node [" + nodeWithoutRemoteClusterClientRole + "] does not have the remote cluster client role enabled"
+        );
+
+        final String nodeWithRemoteClusterClientRole = randomFrom(
+            StreamSupport.stream(localCluster.clusterService().state().nodes().spliterator(), false)
+                .map(DiscoveryNode::getName)
+                .filter(nodeName -> nodeWithoutRemoteClusterClientRole.equals(nodeName) == false)
+                .filter(nodeName -> nodeName.equals(pureDataNode) == false)
+                .collect(Collectors.toList()));
+
+        final SearchResponse resp = localCluster.client(nodeWithRemoteClusterClientRole)
+            .prepareSearch("demo", "cluster_a:prod")
+            .setQuery(new MatchAllQueryBuilder())
+            .setAllowPartialSearchResults(false)
+            .setSize(1000)
+            .get();
+        assertHitCount(resp, demoDocs + prodDocs);
+    }
+
+    public void testProxyConnectionDisconnect() throws Exception {
+        assertAcked(client(LOCAL_CLUSTER).admin().indices().prepareCreate("demo"));
+        indexDocs(client(LOCAL_CLUSTER), "demo");
+        final String remoteNode = cluster("cluster_a").startDataOnlyNode();
+        assertAcked(client("cluster_a").admin().indices().prepareCreate("prod")
+            .setSettings(Settings.builder().put("index.routing.allocation.require._name", remoteNode)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0).build()));
+        indexDocs(client("cluster_a"), "prod");
+        SearchListenerPlugin.blockQueryPhase();
+        try {
+            PlainActionFuture<SearchResponse> future = new PlainActionFuture<>();
+            SearchRequest searchRequest = new SearchRequest("demo", "cluster_a:prod");
+            searchRequest.allowPartialSearchResults(false);
+            searchRequest.setCcsMinimizeRoundtrips(false);
+            searchRequest.source(new SearchSourceBuilder().query(new MatchAllQueryBuilder()).size(1000));
+            client(LOCAL_CLUSTER).search(searchRequest, future);
+            SearchListenerPlugin.waitSearchStarted();
+            disconnectFromRemoteClusters();
+            assertBusy(() -> assertTrue(future.isDone()));
+            configureAndConnectsToRemoteClusters();
+        } finally {
+            SearchListenerPlugin.allowQueryPhase();
+        }
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins(String clusterAlias) {
+        if (clusterAlias.equals(LOCAL_CLUSTER)) {
+            return super.nodePlugins(clusterAlias);
+        } else {
+            final List<Class<? extends Plugin>> plugins = new ArrayList<>(super.nodePlugins(clusterAlias));
+            plugins.add(SearchListenerPlugin.class);
+            return plugins;
```

Comment:
NIT: we have `CollectionUtils.appendToCopy(super.nodePlugins(clusterAlias), SearchListenerPlugin.class);` to make this stuff one line

Reply:
Nice, see https://github.com/elastic/elasticsearch/pull/65697/commits/3e262a7349647bb6b3c48ab1b840d88396aa08ce